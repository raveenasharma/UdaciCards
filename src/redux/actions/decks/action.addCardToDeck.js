import {ADD_CARD_TO_DECK} from './types'
export const addCardToDeck = payload => {
  return {type: ADD_CARD_TO_DECK, payload}
}

export default addCardToDeck
