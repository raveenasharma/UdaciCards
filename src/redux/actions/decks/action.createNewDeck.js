import {CREATE_NEW_DECK} from './types'
const createNewDeck = payload => {
  return {type: CREATE_NEW_DECK, payload}
}

export default createNewDeck
