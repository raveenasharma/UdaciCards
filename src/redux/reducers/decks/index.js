import {
  CREATE_NEW_DECK,
  RENAME_DECK,
  DELETE_DECK,
  DELETE_ALL_DECKS,
  RESTORE_DEFAULT_DECKS,
  ADD_CARD_TO_DECK
} from '../../actions/decks/types'
import addCardToDeck from './reducer.addCardToDeck'
import createNewDeck from './reducer.createNewDeck'
import deleteAllDecks from './reducer.deleteAllDecks'
import deleteDeck from './reducer.deleteDeck'
import renameDeck from './reducer.renameDeck'
import restoreDefaultDecks from './reducer.restoreDefaultDecks'

export default function decks (state = {}, action) {
  switch (action.type) {
    case CREATE_NEW_DECK:
      return createNewDeck(state, action)
    case RENAME_DECK:
      return renameDeck(state, action)
    case DELETE_DECK:
      return deleteDeck(state, action)
    case DELETE_ALL_DECKS:
      return deleteAllDecks(state, action)
    case RESTORE_DEFAULT_DECKS:
      return restoreDefaultDecks(state, action)
    case ADD_CARD_TO_DECK:
      return addCardToDeck(state, action)
    default:
      return state
  }
}
