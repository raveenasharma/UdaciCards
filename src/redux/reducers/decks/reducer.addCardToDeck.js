const addCardToDeck = (state, action) => {
  const {card, deckName} = action.payload
  const updatedDeck = {...state[deckName]}
  updatedDeck.questions = [...state[deckName].questions]
  updatedDeck.questions.push(card)

  return {
    ...state,
    [deckName]: updatedDeck
  }
}

export default addCardToDeck
