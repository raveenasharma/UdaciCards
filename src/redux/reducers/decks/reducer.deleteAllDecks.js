const deleteAllDecks = (state, action) => {
  return {}
}

export default deleteAllDecks
