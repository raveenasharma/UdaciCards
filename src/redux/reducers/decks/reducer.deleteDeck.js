const deleteDeck = (state, action) => {
  const {name} = action.payload
  const newState = {...state}
  delete newState[name]
  return newState
}

export default deleteDeck
