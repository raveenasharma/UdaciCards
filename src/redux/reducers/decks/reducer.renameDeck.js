const renameDeck = (state, action) => {
  const {name, newName} = action.payload
  const deck = {...state[name], title: newName}
  const newState = {...state}
  delete newState[name]
  return {
    ...newState,
    [newName]: deck
  }
}

export default renameDeck
