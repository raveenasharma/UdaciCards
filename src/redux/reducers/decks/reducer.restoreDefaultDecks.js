import initialState from './initial-state'

const restoreDefaultDecks = (state, action) => {
  return {...state, ...initialState}
}

export default restoreDefaultDecks
