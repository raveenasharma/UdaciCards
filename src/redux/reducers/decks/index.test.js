/* global describe, test, expect */
import deepFreeze from 'deep-freeze'
import initialState from './initial-state'
import decks from './'
import {
  CREATE_NEW_DECK,
  RENAME_DECK,
  DELETE_DECK,
  DELETE_ALL_DECKS,
  RESTORE_DEFAULT_DECKS,
  ADD_CARD_TO_DECK
} from '../../actions/decks/types'

describe('Deck reducer', () => {
  test('createNewDeck', () => {
    const state = deepFreeze({...initialState})
    const result = decks(state, {
      type: CREATE_NEW_DECK,
      payload: {name: 'Test Deck'}
    })
    expect(result).toMatchSnapshot()
  })

  test('renameDeck', () => {
    const state = deepFreeze({...initialState})
    const result = decks(state, {
      type: RENAME_DECK,
      payload: {name: 'React', newName: 'Test Deck'}
    })
    expect(result).toMatchSnapshot()
  })

  test('deleteDeck', () => {
    const state = deepFreeze({...initialState})
    const result = decks(state, {
      type: DELETE_DECK,
      payload: {name: 'React'}
    })
    expect(result).toMatchSnapshot()
  })

  test('deleteAllDecks', () => {
    const state = deepFreeze({...initialState})
    const result = decks(state, {type: DELETE_ALL_DECKS})
    expect(result).toMatchSnapshot()
  })

  test('restoreDefaultDecks', () => {
    const state = {}
    const result = decks(state, {type: RESTORE_DEFAULT_DECKS})
    expect(result).toMatchSnapshot()
  })

  test('addCardToDeck', () => {
    const state = deepFreeze({...initialState})
    const result = decks(state, {
      type: ADD_CARD_TO_DECK,
      payload: {
        deckName: 'React',
        card: {
          question: 'question',
          answer: 'answer'
        }
      }
    })
    expect(result).toMatchSnapshot()
  })
})
