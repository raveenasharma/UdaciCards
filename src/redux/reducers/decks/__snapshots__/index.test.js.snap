// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck reducer addCardToDeck 1`] = `
Object {
  "JavaScript": Object {
    "questions": Array [
      Object {
        "answer": "The combination of a function and the lexical environment within which that function was declared.",
        "question": "What is a closure?",
      },
    ],
    "title": "JavaScript",
  },
  "React": Object {
    "questions": Array [
      Object {
        "answer": "A library for managing user interfaces",
        "question": "What is React?",
      },
      Object {
        "answer": "The componentDidMount lifecycle event",
        "question": "Where do you make Ajax requests in React?",
      },
      Object {
        "answer": "answer",
        "question": "question",
      },
    ],
    "title": "React",
  },
}
`;

exports[`Deck reducer createNewDeck 1`] = `
Object {
  "JavaScript": Object {
    "questions": Array [
      Object {
        "answer": "The combination of a function and the lexical environment within which that function was declared.",
        "question": "What is a closure?",
      },
    ],
    "title": "JavaScript",
  },
  "React": Object {
    "questions": Array [
      Object {
        "answer": "A library for managing user interfaces",
        "question": "What is React?",
      },
      Object {
        "answer": "The componentDidMount lifecycle event",
        "question": "Where do you make Ajax requests in React?",
      },
    ],
    "title": "React",
  },
  "Test Deck": Object {
    "questions": Array [],
    "title": "Test Deck",
  },
}
`;

exports[`Deck reducer deleteAllDecks 1`] = `Object {}`;

exports[`Deck reducer deleteDeck 1`] = `
Object {
  "JavaScript": Object {
    "questions": Array [
      Object {
        "answer": "The combination of a function and the lexical environment within which that function was declared.",
        "question": "What is a closure?",
      },
    ],
    "title": "JavaScript",
  },
}
`;

exports[`Deck reducer renameDeck 1`] = `
Object {
  "JavaScript": Object {
    "questions": Array [
      Object {
        "answer": "The combination of a function and the lexical environment within which that function was declared.",
        "question": "What is a closure?",
      },
    ],
    "title": "JavaScript",
  },
  "Test Deck": Object {
    "questions": Array [
      Object {
        "answer": "A library for managing user interfaces",
        "question": "What is React?",
      },
      Object {
        "answer": "The componentDidMount lifecycle event",
        "question": "Where do you make Ajax requests in React?",
      },
    ],
    "title": "Test Deck",
  },
}
`;

exports[`Deck reducer restoreDefaultDecks 1`] = `
Object {
  "JavaScript": Object {
    "questions": Array [
      Object {
        "answer": "The combination of a function and the lexical environment within which that function was declared.",
        "question": "What is a closure?",
      },
    ],
    "title": "JavaScript",
  },
  "React": Object {
    "questions": Array [
      Object {
        "answer": "A library for managing user interfaces",
        "question": "What is React?",
      },
      Object {
        "answer": "The componentDidMount lifecycle event",
        "question": "Where do you make Ajax requests in React?",
      },
    ],
    "title": "React",
  },
}
`;
